torch/csrc/jit/graph_executor.cpp

```diff
@@ -638,11 +640,26 @@ struct GraphExecutorImpl {
     CheckInplace(graph);
   }
 
+  void runPostdiffOptimization(std::shared_ptr<Graph>& graph){
+    // Run post differentiation optimization passes, Autodiff will replace some parts of graph with new graph 
+    // these new graphs usually consists of control flow and miss shape information on nodes, so we run shape
+    // prop and constant prop, as well as loop unrolling in some cases
+    PropagateInputShapes(graph);
+    ConstantPropagation(graph);
```

Comment:
This should call peephole optimization as well i think.

Reply:
`runOptimization` have bunch of other passes that we don't need to run, just want to minimize the passes that needs to be run, plus it does not run `propagateInputShapes`, so I just make a separate group of passes.

Reply:
Could you call `propagateInputShapes` and then call into `runOptimization`? The downside is that we're now maintaining two separate sets of optimizations, and you might miss optimizations that you think you didn't need (as with prim::is_cuda because PeepholeOptimize wasn't called)

Reply:
sounds legit. the only concern i was having is that this will increase our compilation time quite a bit, that's why I want separate the passes in the first place. But given that we are potentially missing bunch of optimizations, I will change it to just run the optimization :)